components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/workflow/HttpWorkflowResponse.java

```diff
@@ -19,22 +19,34 @@
 
 import org.json.simple.JSONObject;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * HTTP workflow response. Can be used to pass workflow complete redirections
  */
 public class HttpWorkflowResponse extends AbstractWorkflowResponse {
 
     private String redirectUrl = "";
-    private String redirectConfirmationMsg = null;
+    private String redirectConfirmationMsg = "";
     private JSONObject jsonPayloadObj = new JSONObject();
-    private JSONObject additionalParameters = new JSONObject();
+    private Map<String, String> additionalParameters = new HashMap<>();
 
     @Override
     @SuppressWarnings("unchecked")
     public String getJSONPayload() {
+        if (additionalParameters != null && !additionalParameters.isEmpty()) {
+            redirectUrl = redirectUrl.concat("?");
+            for (Map.Entry<String, String> entry : additionalParameters.entrySet()) {
+                redirectUrl = redirectUrl.concat(((entry.getKey().concat("=")).concat(entry.getValue())).concat("&"));
+            }
+            //remove tailing "&"
```

Comment:
removal is done in the next line. when url is constructed, '&' is set at the end from the previous loop. that one is removed from this line